inspirehep/modules/records/receivers.py

```diff
@@ -39,33 +41,41 @@ def populate_inspire_document_type(recid, json, *args, **kwargs):
         Adds a field for faceting INSPIRE document type
     """
     inspire_doc_type = []
+    collections = []
     if 'collections' in json:
-        for element in json.get('collections', []):
-            if 'primary' in element and element.get('primary', ''):
-                if element['primary'].lower() == 'published':
-                    inspire_doc_type.append(element['primary'].lower())
-                    break
-                elif element['primary'].lower() == 'thesis':
-                    inspire_doc_type.append('peer reviewed')
-                    break
-                elif element['primary'].lower() == 'book':
-                    inspire_doc_type.append(element['primary'].lower())
-                    break
-                elif element['primary'].lower() == 'bookchapter':
-                    inspire_doc_type.append('book chapter')
-                    break
-                elif element['primary'].lower() == 'proceedings':
-                    inspire_doc_type.append(element['primary'].lower())
-                    break
-                elif element['primary'].lower() == 'conferencepaper':
-                    inspire_doc_type.append('conference paper')
-                    break
-                elif element['primary'].lower() == 'note':
-                    inspire_doc_type.append('note')
-                    break
-                elif element['primary'].lower() == 'report':
-                    inspire_doc_type.append(element['primary'].lower())
-                    break
+        for c in json['collections']:
+            if 'primary' in c and c.get('primary', ''):
+                if isinstance(c['primary'], list):
+                    collections.append(', '.join(c['primary']))
```

Comment:
I am not sure about this. Then inside of collections there would be a value like `hep, review`.

I think if this is fixed in the `dojson` so that there is no list in that field it should be ok.